app/models/user.rb

```diff
@@ -13,4 +13,13 @@ def create_api_access_token!
     api_key = api_keys.create
     APIBearerToken.create(user: self, api_key_id: api_key.id)
   end
+
+  # We need to override this to allow User creation without password
+  def password_required?
+    if new_record?
+      false
+    else
+      !persisted? || !password.nil? || !password_confirmation.nil?
+    end
```

Comment:
This seems contradictory since `!persisted?` is the same as `new_record?` Can we use the opposites like `present?` instead of `!..nil?`?